### PR TITLE
destroy Timer Action

### PR DESCRIPTION
**Pull request by glommer**
### What does this PR do?

This PR allows a TimerAction to be destroyed immediately, without having to spawn a new task for it.
To do that, we perform some cleanups in the timer infrastructure and remove yet another legacy Atomic.
Yay! Timers are starting to look good

### Checklist

[x] I have added unit tests to the code I am submitting
[] My unit tests cover both failure and success scenarios
[] If applicable, I have discussed my architecture
[x] The new code I am adding is formatted using `rustfmt`

### CODE REVIEW

**Comment by glommer** [comment]
because review hasn't started, I forced pushed this branch adding more tests.
I am now confident that the implementation is sane.

**Review by HippoBaro — APPROVED** [review]

File: scipio/src/timer.rs [hunk 1]
```diff
@@ -422,8 +429,37 @@ impl<T: 'static> TimerAction<T> {
...
+    ///         println!("Will not execute this");
+    ///     });
+    ///     action.destroy();
+    ///     action.join().await;
+    /// });
+    /// ```
+    /// [`TimerAction`]: struct.TimerAction
+    /// [`cancel`]: struct.TimerAction.html#method.cancel
+    /// [`join`]: struct.TimerAction.html#method.join
+    pub fn destroy(&self) {
```
Comment (HippoBaro) [inline scipio/src/timer.rs:460]: The distinction between `destroy` and `cancel` seems subtle. What's the difference between calling `destroy()` vs calling `cancel().detach()`? Both are non-blocking.
Maybe you want to make the distinction clearer and why I'd want one over the other.

Comment (glommer) [reply scipio/src/timer.rs:460]: The difference is indeed subtle. Behaviourally there is no difference so from a correctness point of view there is no difference.

From a performance point of view there is a difference:
`destroy()` does not generate a new task.
`cancel().detach()` does.

I thought I had mentioned that in the docs. What changes would you suggest to make that clearer ?

Comment (glommer) [reply scipio/src/timer.rs:460]: if you don't want to wait `destroy` is always preferable. So maybe we just need to say that explicitly ?

**Comment by glommer** [comment]
I added a commit on top to fix the documentation because some of the examples were misleading as I have noticed later.
Because some of the examples were preexisting, that's a patch on top.
Merging this version
